indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -259,9 +259,40 @@ def _process(self):
 
 
 class RHUpdateBooking(RHBookingBase):
+    def _should_split(self, new_start_dt, new_end_dt):
+        today = date.today()
+        is_ongoing_booking = new_start_dt.date() < today
+        old_start_time, old_end_time = self.booking.start_dt.time(), self.booking.end_dt.time()
+        times_changed = new_start_dt.time() != old_start_time or new_end_dt.time() != old_end_time
+        return is_ongoing_booking and times_changed
+
+    def split_bookings(self, booking_data):
+        occurrences = self.booking.occurrences
+        room = self.booking.room
+        is_prebooking = not self.booking.is_accepted
+        occurrences_to_cancel = filter(lambda r: r.start_dt.date() >= date.today(), occurrences)
+        new_start_dt = datetime.combine(occurrences_to_cancel[0].start_dt.date(), booking_data['start_dt'].time())
+        for occurrence_to_cancel in occurrences_to_cancel:
+            occurrence_to_cancel.cancel(session.user)
```

Comment:
Do we really want to send emails for each canceled occurrence here? Seems very spammy. Maybe cancel with `silent=True` here and a proper edit log entry for the booking modification.